### DISCUSSION TITLE
Impossible to build a project with quarkus-kind in the classpath if kind not available?

### DISCUSSION THREAD

**Discussion by PierreBtz** [quarkusio/quarkus#38306] [Q&A] [answered]
Hey folks!

Quick question before I open an issue. I have a project using the `quarkus-kind` to deploy locally in a kind cluster, all good there I like it :)

Now for the issue, when running on the CI, there is no kind available, which shouldn't be a problem since I don't want to deploy. However, I think that `quarkus-kind` unconditionally tries to load the image built in kind, even if `quarkus.kubernetes.deploy` is false, as evidenced by https://github.com/quarkusio/quarkus/blob/8bf9f63815b0f2f6dee9fae3db1ee2e306573ee3/extensions/kubernetes/kind/deployment/src/main/java/io/quarkus/kind/deployment/KindProcessor.java#L124-L130.

I looked around in https://quarkus.io/guides/all-config but didn't find any property to modify this behavior. Am I missing something? Would it make sense to skip `kind load...` if `quarkus.kubernetes.deploy` is false? Or have a different settings to deactivate the load? Or catch the exception and just slip a warning "couldn't load the image into the kind cluster"?

The only workaround I see is to hide the dependency behind a profile in Maven, obviously I don't like it :)

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
I don't think this is the proper behavior, it does sound like a bug to me

**Reply by PierreBtz** [reply]
Noted! Let me build a reproducer and sending an creating an issue !

**Comment by PierreBtz** [comment]
Link https://github.com/quarkusio/quarkus/issues/38322
